src/trigrams/utils.rs

```diff
@@ -33,7 +33,7 @@ fn trigram_occurances_to_positions(
         .into_iter()
         .map(|(trigram, count)| (count, trigram))
         .collect();
-    count_vec.sort_by(|a, b| b.cmp(a));
+    count_vec.sort_unstable_by(|a, b| b.cmp(a));
```

Comment:
`by_key` ?

Reply:
It will surely be faster to use a vec and sort it than using a binary heap here, the amount of cmp will be highly reduced. I like the function rewrite you propose ✅

Reply:
> @greyblake does this sort could be equivalent to below code?

I'd say rather no than yes.
Back in the days, there were some really hard to reproduce tests that would non-deterministically fail.
It was fixed in 30d142d2293c3c232ce4f8fee9ba86d1f49263c1
but now, try to recall what was exactly there I blame myself for not documenting that enough :D